### PR TITLE
CSIDriverRegistrar e2e cleanups

### PR DESCRIPTION
**Pull request by jarrpa**
This is a follow-up to #73677 to fix some minor bugs.

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/csi_volumes.go [hunk 1]
```diff
@@ -233,12 +233,12 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
...
 				}
 
-				driver = drivers.InitMockCSIDriver(config, test.driverExists, test.driverAttachable, nil)
+				driver = drivers.InitMockCSIDriver(config, test.deployDriverCRD, test.driverAttachable, nil)
 				driver.CreateDriver()
+				defer driver.CleanupDriver()
 
-				if test.driverExists {
+				if test.deployDriverCRD {
 					defer destroyCSIDriver(csics, driver)
```
Comment (pohly) [inline test/e2e/storage/csi_volumes.go]: Is the order relevant here?

Previously, `destroyCSIDriver` ran first, then `driver.CleanupDriver`. Now it is the other way around.

Comment (jarrpa) [reply test/e2e/storage/csi_volumes.go]: Order doesn't entirely matter, but conceptually this new order is cleaner. We should delete the CSIDriver CRD before deleting the actual driver pods.

**Comment by msau42** [comment]
/ok-to-test
/approve
/lgtm
/test pull-kubernetes-e2e-gce-alpha-features

/hold 
until alpha suite passes

**Pull request merged**